sdks/java/java8tests/src/test/java/com/google/cloud/dataflow/sdk/transforms/CombineJava8Test.java

```diff
@@ -131,4 +138,31 @@ public void testCombinePerKeyInstanceMethodReference() {
         KV.of("c", 4));
     pipeline.run();
   }
+
+  /**
+   * Tests that we can serialize Combine fns constructed from a lambda. Lambdas can be problematic
+   * because the {@link Class} object is synthetic and cannot be deserialized.
+   */
+  @Test
+  public void testLambdaSerialization() {
```

Comment:
I like the test, but I think we should still have display data. Either:
(1) Mark the class transient so its not involved in serialization (although then we may have problems with places where we clone via serialization)
(2) Introduce some kind of "ClassValue" object, that can be serialized (by name + simple name).